storage_server/report_server.py

```diff
@@ -307,9 +308,19 @@ def storeReportInfo(self,
         path_start = path_ids[0].id if len(path_ids) > 0 else None
 
         suppressed = False
+        source_file = self.session.query(File).get(file_id)
+        source_file_path, source_file_name = ntpath.split(source_file.filepath)
+
         supp = self.session.query(SuppressBug) \
-                           .filter(and_(SuppressBug.run_id == action.run_id,
-                                        SuppressBug.hash == bug_hash)) \
+                           .filter(
+                               or_(and_(SuppressBug.run_id == action.run_id,
+                                        SuppressBug.hash == bug_hash,
+                                        SuppressBug.type == bug_hash_type,
+                                        SuppressBug.file_name == source_file_name),
+                                   and_(SuppressBug.run_id == action.run_id,
+                                        SuppressBug.hash == bug_hash,
+                                        SuppressBug.type == bug_hash_type,
+                                        SuppressBug.file_name == ''))) \
```

Comment:
Please double check if the file_name column is None or empty string after a migration